token-service/src/main/java/org/p2p/token/service/repository/configurator/PercentDifferenceToShowByPriceConfigurator.kt

```diff
@@ -0,0 +1,35 @@
+package org.p2p.token.service.repository.configurator
+
+import java.math.BigDecimal
+import org.p2p.core.token.Token
+import org.p2p.core.token.TokenMetadataExtension
+
+private const val RULE_BY_COUNT_OF_TOKENS = "byCountOfTokensValue"
+
+class PercentDifferenceToShowByPriceConfigurator(
+    private val extensions: TokenMetadataExtension,
+    private val token: Token.Active
+) : TokenConfigurator<Token.Active> {
+
+    override fun config(): Token.Active {
+        val ruleOfProcessing = extensions.ruleOfProcessingTokenPriceWs
+
+        // 1. Select token fiat
+        val tokenFiatValue = token.total
+        // 2. Select token rate
+        val tokenRate = token.rate ?: BigDecimal.ONE
+        // 3. Calculate token value in USD
+        val tokenTotalInUsd = tokenRate * tokenFiatValue
+        // 4. Calculate C value
```

Comment:
what? what is C?

Reply:
I don't know, I took this from description for task )